march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_subgait.py

```diff
@@ -126,3 +171,33 @@ def get_mirror(self, key_1, key_2):
 
         return ModifiableSubgait(mirrored_joints, self.duration, self.gait_type, self.gait_name, mirrored_subgait_name,
                                  self.version, self.description)
+
+    def set_gait_type(self, gait_type):
+        self.gait_type = str(gait_type)
+
+    def set_gait_name(self, gait_name):
+        self.gait_name = gait_name
+
+    def set_description(self, description):
+        self.description = str(description)
+
+    def set_version(self, version):
+        self.version = version
+
+    def set_subgait_name(self, subgait_name):
+        self.subgait_name = subgait_name
+
+    def set_duration(self, duration, rescale=False):
+        for joint in self.joints:
+            # Loop in reverse to avoid out of bounds errors while deleting.
+            for setpoint in reversed(joint.setpoints):
+                if rescale:
+                    setpoint.time = duration * setpoint.time / self.duration
+                else:
+                    if setpoint.time > duration:
+                        joint.setpoints.remove(setpoint)
```

Comment:
You could also write this in a list comprehension
```python
joint.setpoints = [ s.time = duration * s.time / self.duration for s in joint.setpoints if s.time <= duration ]
```

Reply:
Nope. That shit don't work. You can't assign a value in a list comprehension. Removing the loop would require some very fancy and annoying coding, which will not make this any clearer, so I will leave it like this.